apps/storefront/src/pages/QuoteDraft/index.tsx

```diff
@@ -504,44 +503,38 @@ function QuoteDraft({ setOpenPage }: PageProps) {
   };
 
   /**
-   * Clone address and compare with selectedAddress to decide if addressId can be reused.
+   * Clone address and compare with masterCopy to decide if addressId can be reused.
    *
-   * @param address - Address with optional selectedAddress for comparison
+   * @param address - Address with optional masterCopy for comparison
    * @returns Cloned address, potentially with reused addressId if no changes detected
    *
    * Cases:
-   * 1. No Selected Address: return cloned address
-   * 2. Selected Address copy exists and address unchanged: return cloned address with addressId
-   * 3. Selected Address copy exists but address changed: return cloned address without addressId
+   * 1. No Master Copy: return cloned address
+   * 2. Master Copy exists and address unchanged: return cloned address with addressId
+   * 3. Master Copy exists but address changed: return cloned address with addressId set to 0
    */
-  const cloneAddressWithId = (address: AddressWithSelected): ShippingAddress | BillingAddress => {
-    const selectedAddress = address?.selectedAddress;
-
-    if (!selectedAddress) {
-      return cloneDeep(omit(address, ['selectedAddress']));
-    }
-
-    const normalizedSelectedAddress = {
-      ...selectedAddress,
-      companyName: selectedAddress.company || '',
-    };
+  const cloneAddressWithId = (address: AddressWithMasterCopy): ShippingAddress | BillingAddress => {
```

Comment:
as I pointed out in the previous PR, you might be better of splitting master and the rest in here:


```suggestion
  const cloneAddressWithId = ({ masterCopy, ...address }: AddressWithMasterCopy): ShippingAddress | BillingAddress => {
```